Unify sui, wallet, key-tool, sui-move binaries

patrickkuo opened the pull request:
Changes:
* unified sui, wallet, key-tool, sui-move binaries
* wallet functionalities is accessible via `./sui client`
* key-tool functionalities is accessible via `./sui keytool`
* Sui-move functionalities is accessible via `./sui move`
* interactive console is accessible via `./sui console`
* I haven't change sui `start`, `genesis`, and `network` command, we might want to group them into `network` in future PR (e.g. `./sui network genesis` and `./sui network start`)
* Moved `rpc-server` to `sui-gateway` crate
* moved benchmark bins to sui-benchmark crate
* crate dependencies clean up
* Renamed wallet to Sui CLI client everywhere to disambiguate the CLI tool and the new wallet chrome plugin/ wallet app

WIP:

- [x] Docs

TODOs (subsequence PRs):
* I am planning to do some cleanup on `GatewayType` and remove reliance on GatewayAPI
* Improve keytool, add key import/export and more keystore management functionalities.
* Improve env switching, we should support multiple sui env configurations and allow easy environment switching using flags, e.g. `./sui --devnet client ...` or `./sui --testnet client ...`

bmwill reviewed (approved):
This is an awesome cleanup and should actually improve compile times since we'll be building fewer binaries in total.

patrickkuo commented:
> Oh I forgot, you should probably also update the `sui-tools` docker image build as a part of this change.

!! Thanks, I will change that

patrickkuo commented:
Hi @Clay-Mysten , I have made quite a lot of changes in docs and updated the cli outputs in the old wallet.md file, and also renamed it to cli-client.md, not sure if this will break anything in the doc site, please take a look when you have time :)

patrickkuo commented on crates/sui-gateway/Cargo.toml:
```diff
@@ -10,30 +10,28 @@ edition = "2021"
+sui-json-rpc = { path = "../sui-json-rpc" }
 sui-json-rpc-api = { path = "../sui-json-rpc-api" }
-sui-open-rpc = { path = "../sui-open-rpc" }
+sui-node = { path = "../sui-node" }
```
it is because of `GatewayMetrics`, we might want to move it to sui-gateway instead, I will create an issue to do some further cleanup for Gateway.

Clay-Mysten reviewed (approved):
Hi Patrick,

This all LGTM with minor edits I made directly in commits here. Please note, we must also submit these PRs to update the Dev Portal left nav menu and landing page with the new link:
https://github.com/MystenLabs/sui/pull/2859
https://github.com/MystenLabs/sui/pull/2860

In addition, I want to confirm the binaries I just listed in the install docs are truly going away as you remove them here:
https://docs.sui.io/build/install#binaries
---
bench - local test and benchmark of the Sui authorities.
bench_configure - creates the config files for distributed benchmarking.
key_tool - utility for generating keys and addresses encoded as Base64 bytes.
remote_load_generator - enables distributed benchmark of the Sui authorities on remote machines.
---

Finally, I assume you and others have tested all of this thoroughly. Please let me know how else I can help. Kind regards.

patrickkuo commented:
> In addition, I want to confirm the binaries I just listed in the install docs are truly going away as you remove them here: https://docs.sui.io/build/install#binaries
> 
> bench - local test and benchmark of the Sui authorities. bench_configure - creates the config files for distributed benchmarking. key_tool - utility for generating keys and addresses encoded as Base64 bytes. remote_load_generator - enables distributed benchmark of the Sui authorities on remote machines.

I have changed the install cmd to 
`cargo install --locked --git https://github.com/MystenLabs/sui.git --branch "devnet" sui sui-json-rpc`
which will only install `sui` and `rpc-server` binaries; `bench`, `bench-configure` and `remote_load_generator` is not relevant to most user so they are no longer included in the installation. `key-tool` has been integrated into the `sui` binary.

> Finally, I assume you and others have tested all of this thoroughly. Please let me know how else I can help. Kind regards.

Thanks for the help with the docs @Clay-Mysten !!!

Clay-Mysten commented:
@patrickkuo here is the announcement PR:
https://github.com/MystenLabs/sui/pull/2867
